cloudformation/calculation-athena.yaml

```diff
@@ -8,6 +8,52 @@ Resources:
       DatabaseInput:
         Name: !Sub ${AWS::StackName}-calculations
 
+  MonthlyExtractTable:
+    Type: AWS::Glue::Table
+    Properties:
+      CatalogId: !Ref AWS::AccountId
+      DatabaseName: !Ref CalculationsDB
+      TableInput:
+        Name: btm_monthly_extract
+        TableType: EXTERNAL_TABLE
+        Parameters:
+          has_encrypted_data: true
+          storage.location.template: !Sub s3://${StorageBucket}/btm_extract_data/full-extract.json
+        StorageDescriptor:
+          Columns:
+            - Name: vendor_id
+              Type: string
+            - Name: vendor_name
+              Type: string
+            - Name: service_name
+              Type: string
+            - Name: year
+              Type: string
+            - Name: month
+              Type: string
+            - Name: billing_price_formatted
+              Type: string
+            - Name: transaction_price_formatted
+              Type: string
+            - Name: price_difference
+              Type: string
+            - Name: billing_amount_with_tax
+              Type: string
+            - Name: price_difference_percentage
+              Type: double
```

Comment:
The short but unproductive answer is that this table is made to exactly match the old view that Grafana is currently hitting...  As to why the old view was using those types, it's because Grafana can't format fields individually, so the query needed to return the actual formatted values (e.g. "£12,833", using the regex trick you supplied me some weeks ago to put in the commas).

The price_difference_percentage actually has the '%' sign added to it on the Grafana side, but also has some magic number values that indicate things like "Invoice data missing".  This field probably could also be formatted in the query for consistency, but as it's a hack to get around Grafana limitations, I'm hoping we won't have to use it for long.